src/Raven.Server/Web/System/TcpConnectionInfoHandlerForDatabase.cs

```diff
@@ -11,12 +11,25 @@ public class TcpConnectionInfoHandlerForDatabase : DatabaseRequestHandler
         [RavenAction("/databases/*/info/tcp", "GET", AuthorizationStatus.ValidUser, EndpointType.Read, DisableOnCpuCreditsExhaustion = true)]
         public async Task Get()
         {
+            var forExternalUse = CanConnectViaPrivateTcpUrl() == false;
+
             using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
             await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
             {
-                var output = Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl());
+                var output = Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl(), forExternalUse);
                 context.Write(writer, output);
             }
         }
+
+        private bool CanConnectViaPrivateTcpUrl()
+        {
+            var senderUrl = GetStringQueryString("senderUrl", false);
+            if (string.IsNullOrEmpty(senderUrl))
+                return true;
```

Comment:
if there is no `senderUrl`, I would expect to connect via the public url not private one, no?

Reply:
I thought that if `senderUrl` is `null/empty` we keep the previous behavior and try to connect to the private one first...